Merge upstream releases v4.0.0, v4.0.1, and v4.0.2

aaronskiba opened the pull request:
[This](https://docs.google.com/spreadsheets/d/1_BPX0Iix3YyHMNAWAIrdRYtPhKh4SW19IXsp2AKZQmg/edit#gid=0) is a link to a Google Sheets doc with very rough notes regarding how I resolved various conflicts while merging the three releases.

Fixes #470
Fixes #465
Fixes #430
Fixes #416
Fixes #414

benjaminfaure committed:
Updated Ruby version to 2.7.6

benjaminfaure committed:
Updated Ruby version in github workflow files

benjaminfaure committed:
Rubocoped some files

briri committed:
tweak ruby version designation in gemfile to just specify major.minor version

briri committed:
ran bundle install and Version: 2.1.1

Usage: spring COMMAND [ARGS]

Commands for Spring itself:

  binstub         Generate Spring based binstubs. Use --all to generate a binstub for all known commands. Use --remove to revert.
  help            Print available commands.
  server          Explicitly start a Spring server in the foreground
  status          Show current status.
  stop            Stop all Spring processes for this project.

Commands for your application:

  rails           Run a rails command. The following sub commands will use Spring: console, runner, generate, destroy, test.
  rake            Runs the rake command
  rspec           Runs the rspec command to rebuild binstubs for the core rails components

briri committed:
discovered issue running specs on mysql. new version of mysqld considers uniqueness to be case sensitive, so added case insensitivity flag to models and specs. Also added indexes for these

briri committed:
finally fixed autocomplete/typeahead helper for specs

briri committed:
rubocop fixes and fix for index names in migration

briri committed:
fix for outdated db schema

benjaminfaure committed:
Removed case_sensitivity test on region abbreviation as this field is no longer case sentitive

briri committed:
removed db migration that added unqiueness indexes

briri committed:
comment out rspec migration step

briri committed:
updated github actions to use the binstubs

briri committed:
try bundle exec instead of bin/ in github workflows

briri committed:
readd case_insensitive check to region

briri committed:
updates to tests

benjaminfaure committed:
Unpegged mimemagic & pegged font-awesome gem to the major version

benjaminfaure committed:
Replaced turbolinks (no longer maintained) by turbo-rails

benjaminfaure committed:
Removed zaru gem and replaced it by built in ActiveStorage::Filename#sanitize method

benjaminfaure committed:
Commented some unused gems

briri committed:
query optomizations based on feedback from Bullet

briri committed:
attempt to fix randomly failing api v1 deserializer test

benjaminfaure committed:
Updated ESLint to V8 & replace eslint-loaded by eslint-plugin-import

benjaminfaure committed:
Removed eslint-action as github has now everything to run ESLint

See : https://github.com/stefanoeb/eslint-action

benjaminfaure committed:
Fixed eslint action

benjaminfaure committed:
Fixed eslint action

benjaminfaure committed:
Removed deprecated option from tinymce config

benjaminfaure committed:
Upgraded jasmine, jasmine-core & karma-jasmine. Fixed one of the autocomplete test

Commit:
DCC Issue 710: The V0 API calls for assign_users and unassign_users to
departments was broken.

DCC Issue https://github.com/DigitalCurationCentre/DMPonline-Service/issues/710

The fix involved making the following changes to the assign_users() and
unassign_users() methods in
app/controllers/api/v0/departments_controller.rb:
- Removed the previous redirect_to in each method and replaced it with
          render users_api_v0_departments_path
- Ensured the @users variable was set as follows
          @users = @user.org.users.includes(:department)
which is the way @users is set in the users() method in the controller.
The Json was called without this value during rendering.

benjaminfaure committed:
Upgraded chart.js to V3

benjaminfaure committed:
Upgrade js-cookie to V3

briri committed:
Merge pull request #3190 from DMPRoadmap/dcc_issue_710_assign_users_to_dept_api_v1

DCC Issue 710: The V0 API calls for assign_users and unassign_users to

briri committed:
replace deprecated 'resolved_paths' with 'additional_paths' in webpacker.yml which was preventing asset compilation

briri committed:
Merge branch 'rails_6' of github.com:DMPRoadmap/roadmap into rails_6

briri committed:
bumped dependencies

briri committed:
added some config setting for babel to deal with warning messages when compiling assets

briri committed:
Merge branch 'development' of github.com:DMPRoadmap/roadmap into development

briri committed:
Merge branch 'development' into rails_6

briri committed:
updated rubocop rule

briri committed:
make rubocop happy

briri committed:
Merge pull request #3188 from DMPRoadmap/rails_6

DO NOT MERGE - Rails 6 upgrade

briri committed:
testing codepipeline

briri committed:
Merge branch 'development' of github.com:DMPRoadmap/roadmap into development

briri committed:
change to REAME for codepipeline demo

Commit:
Bug #3200 - Fix for the V1 API call GET /api/v1/plans returns duplicate plans in some cases.

Fix for issue #3200.

Changes in app/policies/api/v1/plans_policy.rb methods
plans_for_client() and plans_for_user() returns a unique array of plan
ids by finally calling the method uniq() on the ids the methods
previously returned.

briri committed:
Merge pull request #3202 from DMPRoadmap/bug_3200_v1_api_call_get_plans_returns_duplicates

Bug #3200 - Fix for the V1 API call GET /api/v1/plans returns duplica…

briri committed:
updated dependencies and added brakeman exception for EOL rails

briri committed:
make rubocop happy

briri committed:
Merge pull request #3204 from DMPRoadmap/dependency-updates

updated dependencies and added brakeman exception for EOL rails

Commit:
Bug #3203 - Fix for download plans docx being malformed if it has
character combinations lik \0.. or \F.., e,g, as in
 G:\Hematologie\Onderzoek\Studie Team Hematologie\Studies op alfabet\studies\abc\00. DIGITAAL ISF\03. Contracts - Agreements\a. Clinical Trial Agreement.

Fix for issue #3203.

Change:
The htmltoword gem converts html to docx, so we clean the html string by
replacing single backslashes \ by double backslashes \\.

briri committed:
update to prevent users from being able to make all of their guidance groups optional subsets

briri committed:
Merge branch 'master' into development

briri committed:
Merge branch 'issue3199' into development

briri committed:
remove ability to delete the only non-optiona_subset guidance group

briri committed:
Merge branch 'master' into development

briri committed:
updates to prevent deletion of only non-optional subset

briri committed:
Merge branch 'issue3199' into development

briri committed:
tweak to logic when editing guidance_group

briri committed:
Merge branch 'issue3199' into development

briri committed:
tweak to logic when editing guidance_group

briri committed:
Merge branch 'issue3199' into development

briri committed:
tweak to logic when editing guidance_group

briri committed:
Merge branch 'issue3199' into development

briri committed:
tweak to logic when editing guidance_group

briri committed:
Merge branch 'issue3199' into development

briri committed:
added @guidance_groups to update and create so they are available during view render

briri committed:
Merge branch 'issue3199' into development

nicolasfranck committed:
pluck with ampersand returns not id

briri committed:
Merge pull request #3209 from nicolasfranck/development

pluck with ampersand returns not id

briri committed:
Merge pull request #3207 from DMPRoadmap/bug_3203_error_in_docx_xml_when-dowloading_plans

Bug #3203 - Fix for download plans docx being malformed if it has

Sean-Stilwell committed:
Fixing seeding issue for French translation

briri committed:
Merge pull request #3210 from Sean-Stilwell/master

Fixing seeding issue for French translation

briri committed:
updated Identifier model to strip whitespace before appending IdeentifierScheme prefix

briri committed:
updated fix to first convert the value to string before trying to strip whitespace

briri committed:
some accessibility fixes discovered during audit of dmptool. mostly missing labels for form elements

gjacob24 committed:
Fix for issue #3217 regarding template search feature

benjaminfaure committed:
Fixed issue with Plan & Template visibilities

benjaminfaure committed:
Update gems & js packages

briri committed:
Merge pull request #3218 from DMPRoadmap/template_search_issue_3217

Fix for issue #3217 regarding template search feature

briri committed:
Merge pull request #3212 from DMPRoadmap/dmptool-issue379

Contributor form allows user to include whitespace at the beginning of ORCID

pengyin-shan committed:
commented our Capybara.start

briri committed:
Merge pull request #3224 from DMPRoadmap/fix_Capybara

commented our Capybara.start

briri committed:
Merge branch 'development' into fix/plan_visibility

benjaminfaure committed:
Fixed tests by adding default visibility to template & plan test factories

benjaminfaure committed:
Fixed template tests

benjaminfaure committed:
Rubocoped everything

Commit:
Translation.io new sync for rails6 25/10/2022

briri committed:
Merge pull request #3223 from benjaminfaure/fix/plan_visibility

Fix/plan visibility

briri committed:
Merge pull request #3227 from DMPRoadmap/translation_io_25_10_22

Translation.io new sync for rails6 25/10/2022

briri committed:
Merge pull request #3226 from DMPRoadmap/development

Upgrade to v4.0

briri committed:
Merge pull request #3228 from DMPRoadmap/main

Merging main back into development after the 4.0 release

briri committed:
Update README.md

updated versions of dependencies

briri committed:
updated bin/setup so that its simpler to install the system

briri committed:
removed unused files and reverted schema.rb

briri committed:
updated github actions to work with changed sample config files. some rubocop fixes. reran Brakeman

briri committed:
revert some rubocop changes since they were auto-corrected with the wrong version

pengyin-shan committed:
fix YAML safe load capatibility potential error

pengyin-shan committed:
move common config setting to application.rb

briri committed:
Merge branch 'development' into accessibility-fixes

briri committed:
attempt to make setup work with Windows

pengyin-shan committed:
add translation format for feedback notification page

pengyin-shan committed:
add translation format for at_csv.rb

pengyin-shan committed:
fix rubocop

pengyin-shan committed:
update webdriver to 5.2 for a webdriver bug on M-powered Mac

pengyin-shan committed:
add tranlsation formats for usage section

pengyin-shan committed:
move the <p> tag out of translation section

briri committed:
Added missing translation for email that lists new user permissions

briri committed:
fixed issue with the Org links which was causing save errors

briri committed:
switched from db:prepare to db:create

pengyin-shan committed:
add popover for org_profile page

pengyin-shan committed:
update message for public plan definition

pengyin-shan committed:
update csv name for statistics

benjaminfaure committed:
Created editorconfig file

pengyin-shan committed:
remove the extra Affiliation row in csv download

pengyin-shan committed:
change github action ubuntu version to ubuntu-20.04 instead of ubuntu-latest

pengyin-shan committed:
fix routes.rb to replace some double quotes with single quotes

pengyin-shan committed:
Merge pull request #3248 from DMPRoadmap/fix_3247

Fix issue 3247: change ubuntu version in Github Actions to ubuntu-20.04

briri committed:
Merge branch 'development' into accessibility-fixes

briri committed:
fixed typo in research_outputs view

briri committed:
fixed typo in textarea question view

briri committed:
Merge pull request #3213 from DMPRoadmap/accessibility-fixes

Accessibility fixes

briri committed:
Merge branch 'development' into install-update

briri committed:
Merge pull request #3229 from DMPRoadmap/install-update

Updates to bin/setup to make installation easier

briri committed:
Merge pull request #3240 from DMPRoadmap/patch-rails6

Patches to Rails 6

briri committed:
Merge branch 'development' into fix_double_quotes_routes

briri committed:
Merge branch 'development' into remove_extra_affiliation

briri committed:
Merge branch 'development' into editorconfig

briri committed:
Merge branch 'development' into statistics_filename_change

briri committed:
Merge branch 'development' into org_type_tip

briri committed:
Merge branch 'development' into add_translation

briri committed:
Merge branch 'development' into add_translation_feedback_notification

pengyin-shan committed:
Merge branch 'development' into webdriver_update

pengyin-shan committed:
Merge pull request #3236 from DMPRoadmap/webdriver_update

Webdriver update

pengyin-shan committed:
Merge branch 'development' into remove_extra_affiliation

pengyin-shan committed:
Merge branch 'development' into add_translation

pengyin-shan committed:
Merge branch 'development' into add_symbol_to_safe_yaml_list

pengyin-shan committed:
Merge branch 'development' into statistics_filename_change

pengyin-shan committed:
Merge branch 'development' into add_translation_feedback_notification

pengyin-shan committed:
Merge branch 'development' into editorconfig

pengyin-shan committed:
Merge branch 'development' into fix_double_quotes_routes

pengyin-shan committed:
Merge pull request #3246 from DMPRoadmap/remove_extra_affiliation

remove the extra Affiliation row in csv download

pengyin-shan committed:
Merge branch 'development' into org_type_tip

pengyin-shan committed:
Merge pull request #3234 from DMPRoadmap/add_translation_feedback_notification

add translation format for feedback notification page

pengyin-shan committed:
Merge branch 'development' into add_symbol_to_safe_yaml_list

pengyin-shan committed:
Merge pull request #3235 from DMPRoadmap/add_translation

add translation format for at_csv.rb and usage section

pengyin-shan committed:
Merge pull request #3250 from DMPRoadmap/fix_double_quotes_routes

Fix #3249: replace some double quotes to single quotes in config/routes.rb

pengyin-shan committed:
Merge branch 'development' into editorconfig

pengyin-shan committed:
Merge pull request #3243 from benjaminfaure/editorconfig

Added EditorConfig file

pengyin-shan committed:
Merge pull request #3231 from DMPRoadmap/add_symbol_to_safe_yaml_list

fix YAML safe load capatibility potential error

pengyin-shan committed:
bundle, yarn update and translation.io update

pengyin-shan committed:
fix rubocop

pengyin-shan committed:
Merge pull request #3251 from DMPRoadmap/pre-4.0.1-release

bundle, yarn update and translation.io update for 4.0.1 release

pengyin-shan committed:
Merge branch 'main' into development

pengyin-shan committed:
Merge pull request #3252 from DMPRoadmap/development

PENDING to push on Dec 1st: v4.0.1

benjaminfaure committed:
Froze mail gem version

benjaminfaure committed:
Added scss files to editorconfig

pengyin-shan committed:
test CHANGELOG file

pengyin-shan committed:
test CHANGELOG file

pengyin-shan committed:
update danger in gemfile

pengyin-shan committed:
update danger in gemfile

pengyin-shan committed:
add danger figuration

pengyin-shan committed:
add danger figuration

pengyin-shan committed:
fix rubocop

pengyin-shan committed:
fix rubocop

benjaminfaure committed:
Rubocoped Gemfile

pengyin-shan committed:
fix test folder name

pengyin-shan committed:
fix test folder name

pengyin-shan committed:
Merge pull request #3258 from DMPRoadmap/changelog_danger

Add Changelog.md and Danger Github action to code review process

pengyin-shan committed:
sync translation again and update Danger file

pengyin-shan committed:
use failure instead of fail to aovid rubocop complain

pengyin-shan committed:
add exception message for translation.io changes

pengyin-shan committed:
fix Dangerfile

benjaminfaure committed:
Updated CHANGELOG

benjaminfaure committed:
Merge branch 'development' into issue3254

briri committed:
Merge branch 'development' into add_scss_to_editorconfig

pengyin-shan committed:
Merge pull request #3263 from DMPRoadmap/translation_sync_dangerfile_update

[#trivial] Translation sync and Dangerfile update

benjaminfaure committed:
Updated CHANGELOG

briri committed:
Merge branch 'development' into issue3254

gjacob24 committed:
added validation with error message when user enters very large value for file size

gjacob24 committed:
edited CHANGELOG.md to add details of the previous PR

briri committed:
added research outputs to the CSV download of the plan

briri committed:
updated JS that interacts with TinyMCE so that it stops using deprecated setMode function

briri committed:
updated changelog

briri committed:
updated changelog

briri committed:
fix to properly use max per page settings

briri committed:
added changelog entry

briri committed:
updated api v0 to use max_pages config value for api

briri committed:
Merge branch 'development' into org_type_tip

briri committed:
Updated PDF font selector to use Roboto instead of Helvetica since Google no longer has Helvetica

updated changelog

benjaminfaure committed:
Merge branch 'development' into add_scss_to_editorconfig

benjaminfaure committed:
Merge pull request #3275 from DMPRoadmap/research_output_file_size_issue

Fix for issue #3161

benjaminfaure committed:
Merge pull request #3276 from DMPRoadmap/tinymce-update

updated JS that interacts with TinyMCE so that it stops using depreca…

benjaminfaure committed:
Merge branch 'development' into research-outputs-csv

benjaminfaure committed:
Merge branch 'development' into issue3254

benjaminfaure committed:
Merge pull request #3256 from benjaminfaure/add_scss_to_editorconfig

Added scss files to editorconfig

benjaminfaure committed:
Merge pull request #3242 from DMPRoadmap/statistics_filename_change

Change csv file name for statistics from 'Completed' to 'Created'

benjaminfaure committed:
Merge pull request #3241 from DMPRoadmap/org_type_tip

add popover for org_profile page and add explanation for public plan

benjaminfaure committed:
Update CHANGELOG.md

briri committed:
updated paginable concern to fix issue with keyword args

briri committed:
fixed an issue that was causing templates.visibility to get set to organizationally_visible even when the org is a funder

briri committed:
updated change log

briri committed:
fixed typo

briri committed:
make rubocop happy

briri committed:
removed before action from api v0 base_controller. opted to just include functionality in the plans controller

briri committed:
Merge branch 'development' of github.com:DMPRoadmap/roadmap into pagination-fix

benjaminfaure committed:
Added missing show_research_outputs option into as_csv method

benjaminfaure committed:
Merge pull request #3279 from DMPRoadmap/pagination-fix

fix to properly use max per page settings

benjaminfaure committed:
Merge branch 'development' into domo-arigato

benjaminfaure committed:
Merge pull request #3284 from DMPRoadmap/domo-arigato

Switched Google font used in PDFs from Helvetica to Roboto

benjaminfaure committed:
Merge branch 'development' into fix-pagniable-links

benjaminfaure committed:
Merge pull request #3285 from DMPRoadmap/fix-pagniable-links

Fix KeywordArgs issue in Paginable concern

benjaminfaure committed:
Merge branch 'development' into issue3254

benjaminfaure committed:
Merge branch 'development' into research-outputs-csv

benjaminfaure committed:
Merge pull request #3277 from DMPRoadmap/research-outputs-csv

added research outputs to the CSV download of the plan

benjaminfaure committed:
Merge branch 'development' into issue3254

benjaminfaure committed:
Merge pull request #3255 from benjaminfaure/issue3254

Froze mail gem version

benjaminfaure committed:
Merge branch 'development' into fix-template-save

benjaminfaure committed:
Merge pull request #3286 from DMPRoadmap/fix-template-save

Template visibility changed to organizationally_visible

benjaminfaure committed:
Updated translations

benjaminfaure committed:
Updated Ruby & JS dependencies

benjaminfaure committed:
Make rubocop happy

benjaminfaure committed:
Merge pull request #3289 from DMPRoadmap/development

Development

aaronskiba committed:
Merge branch 'upstream-release/v4.0.0' into aaron/merge-upstream-releases-v4.0.0-v4.0.1-v4.0.2

aaronskiba committed:
Remove duplicate code added after v4.0.0 merge

These duplicate lines of code resulted from merging of upstream release v4.0.0 (commit 348bf30841c4eb97b830b6c242f7c1ae3cbe0c9c). (These did not appear as merge conflicts at the time.)

aaronskiba committed:
Configure translation.rb after v4.0.0 merge

Initially, I accepted both changes when merging upstream release v4.0.0 (commit 348bf30841c4eb97b830b6c242f7c1ae3cbe0c9c).

aaronskiba committed:
Comment out logger changes after merging v4.0.0

These now commented out lines were introduced from the merging of upstream release v4.0.0 (commit 348bf30841c4eb97b830b6c242f7c1ae3cbe0c9c)

aaronskiba committed:
Merge branch 'upstream-release/v4.0.1' into aaron/merge-upstream-releases-v4.0.0-v4.0.1-v4.0.2

aaronskiba committed:
Modify mysql.yml with corresponding v4.0.1 file

Copy/pasted corresponding [upstream release 4.0.1 mysql.yml file](https://github.com/DMPRoadmap/roadmap/blob/a803f43715b8174970dfcc5b6a9a103b36efe1e3/.github/workflows/mysql.yml) and then made the following modifications:
- Set action to occur on `push` again as well
- set `- uses: actions/checkout@v2` back to `- uses: actions/checkout@v3` for `# Checkout the repo`
- Put back `node-version: 16` for `# Install node`

aaronskiba committed:
Merge branch 'upstream-release/v4.0.2' into aaron/merge-upstream-releases-v4.0.0-v4.0.1-v4.0.2

aaronskiba committed:
#TEMP Pin rubocop to match upstream release v4.0.2

Set to match [upstream release v4.0.2 rubocop version](https://github.com/DMPRoadmap/roadmap/blob/d95a1a9f07fbc9c313fada692b6f54e1193eb359/Gemfile.lock#L425)

aaronskiba committed:
`bundle update`

aaronskiba committed:
replace 'safe_email' with 'email' for Faker

This addresses a Faker deprecation warning:
https://github.com/faker-ruby/faker/blob/main/doc/default/internet.md#about-faker-ruby-transition-to-be-rfc-2606-compliant

aaronskiba committed:
#TEMP Skip 'User creates a new Plan' test

Put this back after fixing features/ tests with upstream developers.

aaronskiba committed:
Change credentials file used in workflows/ruby.yml

Upstream release v4.0.1 replaced credentials.yml.example with credentials.yml.mysql2 and credentials.yml.postgresql (merge commit 7d5bb2ea7221ab9cf69cbe1e636779bf3ba2022c)

aaronskiba committed:
Merge branch 'deployment-portage' into aaron/merge-upstream-releases-v4.0.0-v4.0.1-v4.0.2

aaronskiba committed:
Fix org_types_tooltip format for translation.io

Reverting a change from commit # 46eba76c979c5ac25d2f9c61ea579874d13dde8a. This commit marks the merging of upstream release v4.0.2

aaronskiba committed:
Update CHANGELOG.md

aaronskiba committed:
Remove commented out logger code

This code was introduced when merging upstream release v4.0.0 (commit # 9421b57fc5fdfa88ff6a6fc02992293a85062c36). Lines 86 and 92-96 were commented-out by me.

aaronskiba committed:
Set up dmproadmap_host secret

aaronskiba committed:
Remove en-GB from translation.io supported locales

aaronskiba committed:
add config.host secret to sandbox, staging, & uat

uat is the only config/environment that doesn't use Rails.application.routes.default_url_options[:host]

200455939-yashu commented on app/models/concerns/exportable_plan.rb:
```diff
@@ -50,6 +51,10 @@ def as_csv(user,
         end
       end
+      csv << []
+      csv << []
```
Lines 54 and 55 are the same!

200455939-yashu commented on app/models/concerns/exportable_plan.rb:
```diff
@@ -183,6 +214,21 @@ def prepare_coversheet_for_csv(csv, _headings, hash)
 
+  # rubocop:disable Metrics/AbcSize
+  def prepare_research_outputs_for_csv(csv, _headings, hash)
+    return false unless hash[:research_outputs].present? && hash[:research_outputs].any?
```
can we directly call `retrun unless` instead of `return false unless` ?

lagoan replied:
I think here we want the false value explicitly instead of nil with just the return statement. This logic could be simplified though, we will be able to help more with this once we are more involved with core development.

200455939-yashu commented on app/models/concerns/exportable_plan.rb:
```diff
@@ -183,6 +214,21 @@ def prepare_coversheet_for_csv(csv, _headings, hash)
+      csv << research_output.values
+    end
+    csv << []
+    csv << []
```
Do we need lines 227 and 228?

lagoan commented on Gemfile:
```diff
@@ -318,29 +326,29 @@ group :ci, :development do
   # RuboCop is a Ruby code style checking and code formatting tool. It aims to enforce
   # the community-driven Ruby Style Guide.
-  gem 'rubocop'
+  gem 'rubocop', '~> 1.44.1' # Pinned to align with upstream v4.0.2
```
We should add another comment on the line above pinning the gem specifying it is for dmp assistant

lagoan commented on Gemfile:
```diff
@@ -318,29 +326,29 @@ group :ci, :development do
 
   # Automatic Rails code style checking tool. A RuboCop extension focused on enforcing
   # Rails best practices and coding conventions.
-  gem 'rubocop-rails'
+  # gem 'rubocop-rails'
 
   # A RuboCop plugin for Rake tasks
-  gem 'rubocop-rake'
+  # gem 'rubocop-rake'
 
   # Code style checking for RSpec files. A plugin for the RuboCop code style enforcing
   # & linting tool.
-  gem 'rubocop-rspec'
+  # gem 'rubocop-rspec'
 
   # Thread-safety checks via static analysis. A plugin for the RuboCop code style
   # enforcing & linting tool.
-  gem 'rubocop-thread_safety'
+  # gem 'rubocop-thread_safety'
```
@aaronskiba do we know why these rubocop gems were commented out?

lagoan commented on config/environments/development.rb:
```diff
@@ -62,13 +72,28 @@
 
 # Used by Rails' routes url_helpers (typically when including a link in an email)
-Rails.application.routes.default_url_options[:host] = "localhost:3000"
+Rails.application.routes.default_url_options[:host] = Rails.application.secrets.dmproadmap_host
```
We had a conversation before about the use of ENV variables. In this case, we could continue to use it since it is for the development environment, we use a lot of these variables at the moment. The production environment is where we handle it differently at this time.

lagoan replied:
@aaronskiba Will add ENV variable backl

lagoan commented on config/environments/production.rb:
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require 'syslog/logger'
+require 'active_support/core_ext/integer/time'
```
We've commented out the code where this is used so we can ignore this as well.

```
# config.active_record.database_selector = { delay: 2.seconds }
```

lagoan commented on config/environments/production.rb:
```diff
@@ -98,6 +99,34 @@
+  # these configuration options.
+  # config.active_record.database_selector = { delay: 2.seconds }
+  # config.active_record.database_resolver = ActiveRecord::Middleware::DatabaseSelector::Resolver
+  # config.active_record.database_resolver_context = ActiveRecord::Middleware::DatabaseSelector::Resolver::Session
```
Is upstream using this configuration ? If not we should remove it as it will only add noise to our config file and at this time I dont see a use case for us for multiple databases.

200455939-yashu reviewed (dismissed):
Awesome Work 😇👏

lagoan reviewed (changes requested):
Looks like good work! We should find time to go through the change together.

aaronskiba committed:
Config host with ENV (not secret) for development

aaronskiba committed:
remove unused 'require' statement + commented code

'require' statement is for the `config.active_record.database_selector = { delay: 2.seconds }` statement, which we are not using.
Commented lines 102-122 are for database switching, which we are also not using.

aaronskiba committed:
Add CHANGELOG.md changes from Release v4.0.2

NOTE: CHANGELOG.md did not exist for upstream until release v4.0.2

aaronskiba committed:
Edit rubocop pinning comment in Gemfile

aaronskiba committed:
put back `end` (fix commit fe4e18131)

aaronskiba committed:
Merge branch 'deployment-portage' into aaron/merge-upstream-releases-v4.0.0-v4.0.1-v4.0.2

aaronskiba committed:
Stop Spring reload in production mode

Resulting code changes from running `bin/spring binstub --all`
This fix idea comes from https://github.com/DMPRoadmap/roadmap/releases/tag/v4.1.1

aaronskiba committed:
Flagshihtzu fix to allow Docker GH action to pass

lagoan commented on CHANGELOG.md:
```diff
@@ -1,5 +1,38 @@
+       - Added scss files to EditorConfig
+       - Change csv file name for statistics from 'Completed' to 'Created'
+       - Added error message and updated saving message for plan writing session to improve user experience
+       
```
For merging this file we can add the changes on the same indentation level as they are changes that come into that release. Nice idea of keeping the merged versions, we should keep that.

lagoan commented on app/views/orgs/_profile_form.html.erb:
```diff
@@ -129,9 +129,10 @@
-        <legend><%= _('Organisation Types') %>
+        <legend>
+          <%= _('Organisation Types') %>
             &nbsp; <a href="#" aria-label="<%= _('Text') %>" data-toggle="tooltip" data-placement="right" title="<%= org_types_tooltip %>">
-             <i class="fas fa-question-circle fa-reverse" ></i>
+            <i class="fas fa-question-circle fa-reverse" ></i>
             <em class="sr-only"></em>
```
I think we were going to verify the tooltips were working. Did we verify this?

lagoan reviewed (changes requested):
Great progress @aaronskiba , this is a huge PR so there are only a few things to look at.

aaronskiba committed:
Edit Alignment of Changelog entries

Keep upstream's changelog entries inline with our own.

lagoan reviewed (approved):
![good job!](https://media.giphy.com/media/xjc9ZGqcMukD1EJo9r/giphy.gif)
